lib/seadsa/DsaGlobal.cc

```diff
@@ -612,7 +612,8 @@ bool BottomUpTopDownGlobalAnalysis::runOnModule(Module &M) {
       errs() << "Started bottom-up + top-down global analysis ... \n");
 
   for (auto &F : M) {
-    if (F.isDeclaration() || F.empty()) continue;
+    if (F.empty() && !m_specGraphInfo.hasSpecFunc(F)) continue;
```

Comment:
pull common condition out. See comment in another commit for the same looking code.